src/main/java/mods/eln/Eln.java

```diff
@@ -1148,134 +1165,161 @@ private void registerElectricalCable(int id) {
-
-            lowVoltageCableDescriptor = desc;
-
-            desc.setPhysicalConstantLikeNormalCable(LVU, LVP(), 0.2 / 20,// electricalNominalVoltage,
-                // electricalNominalPower,
-                // electricalNominalPowerDrop,
-                LVU * 1.3, LVP() * 1.2,// electricalMaximalVoltage,
-                // electricalMaximalPower,
-                20,// electricalOverVoltageStartPowerLost,
-                cableWarmLimit, -100,// thermalWarmLimit, thermalCoolLimit,
-                cableHeatingTime, cableThermalConductionTao// thermalNominalHeatTime,
-                // thermalConductivityTao
-            );
-
-            sixNodeItem.addDescriptor(subId + (id << 6), desc);
-
-            desc = new ElectricalCableDescriptor(name, stdCableRender50V,
-                "For low voltage with high current.", false);
+        double power = 0;
+        double nominalcablevoltage = 0;
+        double nominalcablepower = 0;
+
+        stdCableRender50V = new CableRenderDescriptor("eln", "sprites/cable.png", 1.95f, 0.95f);
+        stdCableRender200V = new CableRenderDescriptor("eln", "sprites/cable.png", 2.95f, 0.95f);
+        stdCableRender800V = new CableRenderDescriptor("eln", "sprites/cable.png", 3.95f, 1.95f);
+        stdCableRender3200V = new CableRenderDescriptor("eln", "sprites/cableVHV.png", 3.95f, 1.95f);
+
+        List<String> voltagelist = new ArrayList<>(Arrays.asList(
+            "Low ",
+            "Medium ",
+            "High ",
+            "Very High "));
+        List<String> qualitylist = new ArrayList<>(Arrays.asList(
+            "Standard ",
+            "Shoddy ",
+            "Quality ",
+            "Perfect "));
+        List<Double> valuelist = new ArrayList<>(Arrays.asList(
+            1d,
+            0.5d,
+            2d,
+            4d));
+        List<String> renderlist = new ArrayList<>(Arrays.asList(
+            "sprites/cable.png",
+            "sprites/cableshoddy.png",
+            "sprites/cablequality.png",
+            "sprites/cableperfect.png",
+            "sprites/cableVHV.png",
+            "sprites/cableVHVshoddy.png",
+            "sprites/cableVHVquality.png",
+            "sprites/cableVHVperfect.png"));
+        List<Float> rendercablesizes = new ArrayList<>(Arrays.asList(
+            1.95f,
+            0.95f,
+            2.95f,
+            0.95f,
+            3.95f,
+            1.95f));
+
+        int renderspriteindex = 0;
+        int renderfloatindex = 0;
+        subId = 3;
+        for(int cablevoltage = 0; cablevoltage <=3; cablevoltage++){
+        for(int cablequality = 0; cablequality <=3; cablequality++) {
+            //while (cablevoltage <= 3) {
+            renderspriteindex = 0;
+            subId++;
+            name = TR_NAME(Type.NONE, qualitylist.get(cablequality) + voltagelist.get(cablevoltage) + "Voltage Cable");
+
+            renderspriteindex = cablequality;
+
+            switch (cablevoltage) {
```

Comment:
I'm very tempted to say "Do this in Kotlin", because the code would be much cleaner. I won't, but I will at some point in the future, if not about this code; you need to learn it anyway.

In particular, read about 'when' expressions [here](http://kotlinlang.org/docs/reference/control-flow.html), right now. You'll love them, and there's no way to forget a break. Switches should be limited to situations where you *want* to leave out breaks.